src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableActorContext.cs

```diff
@@ -0,0 +1,121 @@
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    /// <summary>
+    /// Context object passed to application code executing actor operations.
+    /// </summary>
+    internal class DurableActorContext : DurableCommonContext, IDurableActorContext
+    {
+        private readonly ActorId self;
+
+        public DurableActorContext(DurableTaskExtension config, ActorId actor)
+         : base(config, actor.ActorClass)
+        {
+            this.self = actor;
+        }
+
+        internal SchedulerState State { get; set; }
+
+        internal OperationMessage CurrentOperation { get; set; }
+
+        internal ResponseMessage CurrentOperationResponse { get; set; }
+
+        internal bool Fresh { get; set; }
+
+        internal bool DestructOnExit { get; set; }
+
+        string IDurableActorContext.ActorClass => this.self.ActorClass;
+
+        string IDurableActorContext.Key => this.self.ActorKey;
+
+        ActorId IDurableActorContext.Self => this.self;
+
+        string IDurableActorContext.OperationName
+        {
+            get
+            {
+                this.ThrowIfInvalidAccess();
+                return this.CurrentOperation.Operation;
+            }
+        }
+
+        ILogger IDurableActorContext.Logger => throw new NotImplementedException();
+
+        bool IDurableActorContext.NewlyConstructed
+        {
+            get
+            {
+                this.ThrowIfInvalidAccess();
+                return this.Fresh;
+            }
+        }
+
+        void IDurableActorContext.DestructOnExit()
+        {
+            this.ThrowIfInvalidAccess();
+            this.DestructOnExit = true;
+        }
+
+        TArgument IDurableActorContext.GetOperationContent<TArgument>()
+        {
+            this.ThrowIfInvalidAccess();
+            return this.CurrentOperation.GetContent<TArgument>();
+        }
+
+        object IDurableActorContext.GetOperationContent(Type argumentType)
+        {
+            this.ThrowIfInvalidAccess();
+            return this.CurrentOperation.GetContent(argumentType);
+        }
+
+        IStateView<TState> IDurableActorContext.GetStateAs<TState>(Formatting formatting, JsonSerializerSettings settings)
+        {
+            this.ThrowIfInvalidAccess();
+            if (this.State.CurrentStateView != null)
+            {
+                if (this.State.CurrentStateView is IStateView<TState> view)
+                {
+                    return view;
+                }
+
+                this.State.CurrentStateView.Dispose();
+            }
+
+            var newView = new TypedStateView<TState>(this, formatting, settings);
+            this.State.CurrentStateView = newView;
+            return newView;
+        }
+
+        void IDurableActorContext.Return<TResult>(TResult result)
+        {
+            this.ThrowIfInvalidAccess();
+            this.CurrentOperationResponse.SetResult(result);
+        }
+
+        void IDurableActorContext.Return(object result, Type resultType)
```

Comment:
In my own framework I am always passing the type because it is required for some serialization settings to work correctly (e.g. TypenameHandling.Automatic). However, since this is not how it is done anywhere else in your APIs, nor is that setting really supported, I agree that I should simplify it as you suggest.